crypto/BRCryptoWallet.c

```diff
@@ -903,9 +903,11 @@ cryptoWalletSweeperValidateSupported (BRCryptoNetwork network,
         return CRYPTO_WALLET_SWEEPER_INVALID_ARGUMENTS;
     }
 
-    if (CRYPTO_FALSE == cryptoCurrencyIsIdentical (currency, cryptoWalletGetCurrency (wallet))) {
+    BRCryptoCurrency walletCurrency = cryptoWalletGetCurrency (wallet);
+    if (CRYPTO_FALSE == cryptoCurrencyIsIdentical (currency, walletCurrency)) {
         return CRYPTO_WALLET_SWEEPER_INVALID_ARGUMENTS;
     }
+    cryptoCurrencyGive (walletCurrency);
```

Comment:
The 'give' is missed if the prior statement returns?

Reply:
Good catch. Will fix.